File: app/src/main/java/com/antyzero/smoksmog/ui/screen/start/item/ParticulateViewHolder.java
package com.antyzero.smoksmog.ui.screen.start.item;

import android.content.res.Resources;
import android.view.View;
import android.widget.TextView;

import com.antyzero.smoksmog.R;
import com.antyzero.smoksmog.SmokSmogApplication;
import com.antyzero.smoksmog.settings.Percent;
import com.antyzero.smoksmog.settings.SettingsHelper;
import com.antyzero.smoksmog.ui.view.IndicatorView;

import javax.inject.Inject;

import butterknife.Bind;
import butterknife.ButterKnife;
import pl.malopolska.smoksmog.model.Particulate;

public class ParticulateViewHolder extends ListViewHolder<Particulate> {

    private final Resources resources;

    @Inject
    SettingsHelper settingsHelper;

    @Bind( R.id.textViewName )
    TextView textViewName;
    @Bind( R.id.textViewMeasureDay )
    TextView textViewMeasureDay;
    @Bind( R.id.textViewMeasureHour )
    TextView textViewMeasureHour;
    @Bind( R.id.textViewTimeHour )
    TextView textViewTimeHour;
    @Bind( R.id.textViewTimeDay )
    TextView textViewTimeDay;
    @Bind( R.id.indicatorView )
    IndicatorView indicatorView;

    public ParticulateViewHolder( View itemView ) {
        super( itemView );
        SmokSmogApplication.get( itemView.getContext() ).getAppComponent().inject( this );
        ButterKnife.bind( this, itemView );
        resources = itemView.getContext().getResources();
    }

    @Override
    public void bind( Particulate data ) {
        super.bind( data );

        textViewName.setText( data.getShortName() );
        textViewMeasureDay.setText( resources.getString( R.string.measurment, data.getAverage(), data.getUnit() ) );
        textViewMeasureHour.setText( resources.getString( R.string.measurment, data.getValue(), data.getUnit() ) );

        if ( data.getValue() > data.getNorm() ) {
            textViewTimeHour.setBackgroundResource( R.drawable.shape_oval_iron_border );
        } else {
            textViewTimeHour.setBackgroundResource( R.drawable.shape_oval_iron );
        }

        if ( data.getAverage() > data.getNorm() ) {
            textViewTimeDay.setBackgroundResource( R.drawable.shape_oval_iron_border );
        } else {
            textViewTimeDay.setBackgroundResource( R.drawable.shape_oval_iron );
        }

        if ( Percent.HOUR.equals( settingsHelper.getPercentMode() ) ) {
            indicatorView.setValue( data.getValue() / data.getNorm() );
        } else {
            indicatorView.setValue( data.getAverage() / data.getNorm() );
        }
    }
}

File: app/src/main/kotlin/com/antyzero/smoksmog/ui/screen/start/item/ParticulateViewHolder.kt
package com.antyzero.smoksmog.ui.screen.start.item


import android.content.res.Resources
import android.view.View
import android.view.View.GONE
import android.view.View.VISIBLE
import android.widget.ImageView
import android.widget.TextView
import com.antyzero.smoksmog.BuildConfig
import com.antyzero.smoksmog.R
import com.antyzero.smoksmog.SmokSmogApplication
import com.antyzero.smoksmog.dsl.findViewById
import com.antyzero.smoksmog.settings.Percent
import com.antyzero.smoksmog.settings.SettingsHelper
import com.antyzero.smoksmog.utils.TextUtils
import pl.malopolska.smoksmog.model.Particulate
import pl.malopolska.smoksmog.model.ParticulateEnum
import smoksmog.air.AirQuality
import smoksmog.air.AirQualityIndex
import smoksmog.ui.IndicatorView
import javax.inject.Inject

class ParticulateViewHolder(itemView: View) : ListViewHolder<Particulate>(itemView) {

    private val resources: Resources = itemView.context.resources

    @Inject lateinit var settingsHelper: SettingsHelper

    val textViewName: TextView = findViewById(R.id.textViewName) as TextView
    val textViewMeasureDay: TextView = findViewById(R.id.textViewMeasureDay) as TextView
    val textViewMeasureHour: TextView = findViewById(R.id.textViewMeasureHour) as TextView
    val textViewTimeHour: TextView = findViewById(R.id.textViewTimeHour) as TextView
    val textViewTimeDay: TextView = findViewById(R.id.textViewTimeDay) as TextView
    val indicatorView: IndicatorView = findViewById(R.id.indicatorView) as IndicatorView
    val particulateIndex: ImageView = findViewById(R.id.particulateIndex) as ImageView

    init {
        SmokSmogApplication[itemView.context].appComponent.inject(this)
    }

    override fun bind(data: Particulate) {
        super.bind(data)

        textViewName.text = TextUtils.spannableSubscript(data.shortName)

        textViewMeasureDay.text = resources.getString(R.string.measurment, data.average, data.unit)
        textViewMeasureHour.text = resources.getString(R.string.measurment, data.value, data.unit)

        if (data.value > data.norm && ParticulateEnum.PM25 != data.enum) {
            textViewTimeHour.setBackgroundResource(R.drawable.shape_oval_iron_border)
        } else {
            textViewTimeHour.setBackgroundResource(R.drawable.shape_oval_iron)
        }

        if (data.average > data.norm) {
            textViewTimeDay.setBackgroundResource(R.drawable.shape_oval_iron_border)
        } else {
            textViewTimeDay.setBackgroundResource(R.drawable.shape_oval_iron)
        }

        if (Percent.HOUR == settingsHelper.percentMode) {
            indicatorView.setValue(data.value / data.norm)
        } else {
            indicatorView.setValue(data.average / data.norm)
        }

        if(ParticulateEnum.PM25 == data.enum){
            indicatorView.visibility = GONE
            textViewMeasureDay.visibility = GONE
            textViewTimeDay.visibility = GONE
        }

        particulateIndex.setColorFilter(AirQuality.findByValue(AirQualityIndex.calculate(data)).getColor(context))
    }
}
